### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by yjshen**
The only flaky test remains: `csv_query_window_with_partition_by`.

### CODE REVIEW

File: datafusion/tests/sql.rs [hunk 1]
```diff
@@ -4323,16 +4323,9 @@ async fn test_cast_expressions_error() -> Result<()> {
...
-    match result {
-        Ok(_) => panic!("expected error"),
-        Err(e) => {
-            assert_contains!(e.to_string(),
-                             "Cast error: Cannot cast string 'c' to value of arrow::datatypes::types::Int32Type type"
-            );
-        }
-    }
+    let actual = execute(&mut ctx, sql).await;
+    let expected = vec![vec![""]; 100];
```
Comment (yjshen) [inline datafusion/tests/sql.rs:4327]: cast string to Int32 is possible in arrow2, cast failure will result in NULL

https://github.com/jorgecarleitao/arrow2/blob/main/src/compute/cast/mod.rs#L129

File: datafusion/src/physical_plan/array_expressions.rs [hunk 2]
```diff
@@ -25,21 +25,91 @@ use std::sync::Arc;
...
-fn array_array(arrays: &[&dyn Array]) -> Result<FixedSizeListArray> {
+fn array_array(arrays: &[&dyn Array]) -> Result<ArrayRef> {
     assert!(!arrays.is_empty());
     let first = arrays[0];
     assert!(arrays.iter().all(|x| x.len() == first.len()));
     assert!(arrays.iter().all(|x| x.data_type() == first.data_type()));
 
     let size = arrays.len();
 
-    let values = concat::concatenate(arrays)?;
```
Comment (yjshen) [inline datafusion/src/physical_plan/array_expressions.rs:36]: `array_array` is to zip elements at the same index for each array into a FixedSizeArray, therefore not identical to `concatenate` that chain all arrays end to end.

File: datafusion/src/physical_plan/csv.rs [hunk 3]
```diff
@@ -441,7 +441,7 @@ impl ExecutionPlan for CsvExec {
                 });
 
                 Ok(Box::pin(CsvStream::new(
-                    self.schema.clone(),
+                    self.projected_schema.clone(),
```
Comment (yjshen) [inline datafusion/src/physical_plan/csv.rs:444]: The cause for failed tests `csv_query_window_with_empty_over` and `csv_query_window_with_order_by`. due to wrong schema provided.

File: datafusion/tests/sql.rs (hunk 1 above)
Comment (houqp) [reply datafusion/tests/sql.rs:4327]: cool, filed https://github.com/apache/arrow-datafusion/issues/1056

**Comment by houqp** [comment]
almost there!

**Pull request merged**